### DISCUSSION TITLE
Lesson 6, Parts 11 & 12 - deploying to anvil, adding localhost to metamask

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#439] [General]
If I put the anvil command into my terminal and I used the resulting RPC and private key to add my localhost network to metamask and then I close vscode, do I have to start all over again when I open vscode and type in anvil again and then go to metamask and put in the RPC and private keys for a new localhost network (if I am still working with the same contract)?

Or if I have done it one time, will the localhost work for all the rest of my work on that contract?

I had an issue in the middle of trying to compile and deploy my contract so I closed vscode and opened it again, and I am not sure what part of the process to start with. Right now I am not able to even put in a command to open the correct file in the terminal for some reason...asked a different question for that.

**Comment by PatrickAlphaC** [comment]
You'd have to "reset your metamask" everytime you restart anvil. 

https://support.metamask.io/hc/en-us/articles/360015488891-How-to-clear-your-account-activity-reset-account

**Reply by RamanSB** [reply]
https://docs.metamask.io/wallet/how-to/run-devnet/#reset-your-local-nonce-calculation
